lib/units/provider/index.js

```diff
@@ -14,6 +14,12 @@ import {setDeviceType} from '../../db/api.js'
 
 export default (function(options) {
     const log = logger.createLogger('provider')
+
+    // Check whether the ipv4 address contains a port indication
+    if (options.adbHost.contains(':') && options.adbHost.contains('.')) {
```

Comment:
One might try to specify a fqdn with port.  
I think the dot check is redundant